app/javascript/src/case_contact_autosave.js

```diff
@@ -24,7 +24,9 @@ $('document').ready(() => {
     document.onload = populateForm() // populate the form when the document is loaded
 
     form.onsubmit = event => {
-      window.localStorage.removeItem(formId)
+      if (document.querySelector('.header-flash').textContent.includes('successfully created')) {
```

Comment:
I would like a better way to do this than depending on text generated elsewhere (maybe `object. persisted?`) but it comes with a nice test that will let us refactor, which is great :)